components/external_image/external_image.tsx

```diff
@@ -1,20 +1,25 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import {getImageSrc} from 'utils/post_utils.jsx';
 
-export default class ExternalImage extends React.PureComponent {
-    static propTypes = {
-        children: PropTypes.func.isRequired,
-        enableSVGs: PropTypes.bool.isRequired,
-        hasImageProxy: PropTypes.bool.isRequired,
-        imageMetadata: PropTypes.object,
-        src: PropTypes.string.isRequired,
-    };
+interface ImageMetaData {
+    format?: string;
```

Comment:
I think this `ImageMetaData` it is actually a `PostImage` object type from `mattermost-redux/types/posts` (that doesn't have the `format` field defined. So I think we should add that field into `mattermost-redux` repo.

In any case, I think the fields are not optional. The image metadata can exists or not, but if it exists, it should have all the fields. And actually, there is another field that is not shown here that is the `frame_count` for gif images.

Reply:
I checked `mattermost-redux/types/posts` and I reckon I can add format property in `PostImage ` and can import from there. Thanks for that information.

I reckon I can change width and height fields are not optional since type error only happens in `external_image_test` file when I change those properties are not optional. Hence, I can change teseCase.

But I am so confused whether I have to add format field as optional or not because I checked `post_attatchment_opengraph`(which uses`PostImage` type) but I cannot find any code has format filed for this type.

Reply:
It's not optional. I checked our backend code and we have this structure defined there as well `PostImage` and you can see it has the following fields defined
![image](https://user-images.githubusercontent.com/17804942/98400970-f113ea80-2032-11eb-85be-452f95b1f689.png)
This shows that the we need to add a `format: string` and `frameCount: number` to the PostImage type in redux.

The redux object should be mapping to the object defined in the backend (or in go terms, the struct)